plugins/com.google.cloud.tools.eclipse.appengine.ui/src/com/google/cloud/tools/eclipse/appengine/ui/AppEngineLibrariesSelectorGroup.java

```diff
@@ -107,55 +104,209 @@ private static String getLibraryName(Library library) {
+  private void addDatabinding() {
+    bindingContext = new DataBindingContext(getDisplayRealm());
     for (Button libraryButton : libraryButtons) {
-      Library library = (Library) libraryButton.getData();
-      if (!library.getLibraryDependencies().isEmpty()) {
-        addDatabindingForDependencies(libraryButton);
-      }
+      addDatabindingForButton(libraryButton);
     }
   }
 
-  private void addDatabindingForDependencies(Button libraryButton) {
-    Library library = (Library) libraryButton.getData();
-    for (String libraryId : library.getLibraryDependencies()) {
-      Button dependencyButton = getButtonForLibraryId(libraryId);
-      if (dependencyButton != null) {
-        ISWTObservableValue libraryButtonSelection = WidgetProperties.selection().observe(libraryButton);
-        IObservableValue dependencyButtonSelection =
-            PojoProperties.value(Button.class, "selection").observe(getDisplayRealm(), dependencyButton);
-        IObservableValue dependencyButtonEnablement =
-            PojoProperties.value(Button.class, "enabled").observe(getDisplayRealm(), dependencyButton);
-
-        WritableValue intermediate = new WritableValue(false, Boolean.class);
-        bindingContext.bindValue(libraryButtonSelection, intermediate);
-        bindingContext.bindValue(dependencyButtonSelection, intermediate);
-        bindingContext.bindValue(dependencyButtonEnablement, intermediate,
-                                 new UpdateValueStrategy().setConverter(BooleanConverter.negate()),
-                                 new UpdateValueStrategy().setConverter(BooleanConverter.negate()));
-      }
+  private void addDatabindingForButton(final Button libraryButton) {
+    final Library library = (Library) libraryButton.getData();
+    ISWTObservableValue libraryButtonSelection = WidgetProperties.selection().observe(libraryButton);
+    ISWTObservableValue libraryButtonEnablement = WidgetProperties.enabled().observe(libraryButton);
+    // library selection UI -> model
+    bindingContext.bindValue(libraryButtonSelection,
+                             new NullComputedValue(getDisplayRealm()),
+                             new UpdateValueStrategy().setConverter(new HandleLibrarySelectionConverter(selectedLibraries,
+                                                                                                        library)),
+                             new UpdateValueStrategy(UpdateValueStrategy.POLICY_NEVER));
+    // library selection model -> UI
+    bindingContext.bindValue(libraryButtonSelection, 
+                             new DependentLibrarySelected(getDisplayRealm(),
+                                                          selectedLibraries,
+                                                          library.getId(),
+                                                          true /* resultIfFound */,
+                                                          new ButtonManuallySelected(libraryButton)),
```

Comment:
This `ButtonManuallySelected` class complicates the code.  It took me a good 10 minutes to wrap my head around it.  I think it would be easier to understand if you just subclassed `DepedentLibrarySelected` and override a method.

Is `DependentLibrarySelected` right?  Maybe `IsSelectedOrRequiredLibrary`?